### PR TITLE
Get services for the merged Green Line

### PR DESCRIPTION
**Pull request by alicia-mbta**
#### Summary of changes
**Asana Ticket:** [🐞 Schedule Finder | No scheduled trips found from the merged Green line page](https://app.asana.com/0/555089885850811/1174356662552522)

The logic to retrieve the services for the merged Green line was not receiving the parameter in the right format.
Also tests were added and a pre-processing of trips.<br/>
**Before**:
image

**After**:
image

### CODE REVIEW

File: apps/services/lib/repo.ex [hunk 1]
```diff
@@ -11,6 +13,11 @@ defmodule Services.Repo do
     by_route_id(route_id, params)
   end
 
+  def by_route_id("Green", params) do
+    expanded_green_routes = Enum.join(GreenLine.branch_ids(), ",")
```
Comment (ryan-mahoney) [inline apps/services/lib/repo.ex:17]: @alicia-mbta could this be:
```
GreenLine.branch_ids()
|> Enum.join(",")
|> by_route_id(params)
```

File: apps/site/lib/site_web/channels/vehicle_channel.ex [hunk 2]
```diff
@@ -42,6 +45,14 @@ defmodule SiteWeb.VehicleChannel do
     stop_name = get_stop_name(vehicle.stop_id)
     trip = Schedules.Repo.trip(vehicle.trip_id)
 
+    # Sometimes trip is nil, so to avoid errors, we create a "polished" trip:
+    preprocessed_trip =
+      if trip == nil do
```
Comment (arkadyan) [inline apps/site/lib/site_web/channels/vehicle_channel.ex:50]: This is totally fine, but just to mention another option: `nil` is falsy, so you can also do something like `trip || %{shape_id: nil}`.

File: apps/site/lib/site_web/channels/vehicle_channel.ex [hunk 3]
```diff
@@ -51,14 +62,14 @@ defmodule SiteWeb.VehicleChannel do
           id: vehicle.id,
           icon: "vehicle-bordered-expanded",
           rotation_angle: vehicle.bearing,
-          shape_id: trip.shape_id,
+          shape_id: preprocessed_trip.shape_id,
```
Comment (arkadyan) [inline apps/site/lib/site_web/channels/vehicle_channel.ex:65]: Following up on my comment above, using an inverse of the same trick you can also do something like `trip && trip.shape_id` here and avoid having to make the `preprocessed_trip` in the first place.

File: apps/site/lib/site_web/channels/vehicle_channel.ex [hunk 4]
```diff
@@ -51,7 +54,7 @@ defmodule SiteWeb.VehicleChannel do
           id: vehicle.id,
           icon: "vehicle-bordered-expanded",
           rotation_angle: vehicle.bearing,
-          shape_id: trip.shape_id,
+          shape_id: (trip && trip.shape_id) || nil,
```
Comment (arkadyan) [inline apps/site/lib/site_web/channels/vehicle_channel.ex:57]: `trip && trip.shape_id` will return `nil` if `trip` is `nil`, so the ` || nil` is unecessary.

    iex(1)> trip = nil
    nil
    iex(2)> trip && trip.shape_id
    nil

**Pull request merged**